src/List/List.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './List.scss';
+
+class List extends React.Component {
+  list = this.props.list;
+
+  state = {
+    selectedGoods: [],
+  };
+
+  handleAdd(goodName) {
+    const goodIndex = this.state.selectedGoods.indexOf(goodName);
+
+    if (goodIndex !== -1) {
+      this.state.selectedGoods.splice(goodIndex, 1);
+      this.setState(prevState => ({
+        selectedGoods: prevState.selectedGoods,
+      }));
```

Comment:
так делать нельзя, мутировала стейт и потом записала тот же самый стейт. сейчас у тебя это работает, но с более сложными вещами будут баги. реакт не знает тут о том что стейт изменился. правильно будет сделать так:
```suggestion
      this.setState(prevState => ({
        selectedGoods: prevState.selectedGoods.filter(item => goodName !== item),
      }));
```